### PR TITLE
Unified way of accessing security keywords

### LINKED ISSUE #2099: Unify the way we check for security keywords
NIT: in two other rules we check security keywords in a different approach, it will be nice to unify that (I think the approach that your using in this PR is more readable):

https://github.com/mozilla/bugbot/blob/f6272f4cef19888b28590269079404691c058779/bugbot/rules/crash_small_volume.py#L118

https://github.com/mozilla/bugbot/blob/f6272f4cef19888b28590269079404691c058779/bugbot/rules/security_unhide_dups.py#L51

_Originally posted by @suhaibmujahid in https://github.com/mozilla/bugbot/pull/2098#discussion_r1219524158_

### PR DESCRIPTION
**Pull request by shubham-s-del**
resolves #2099 

## Checklist

- [ ] Type annotations added to new functions
- [ ] Docs added to functions touched in main classes
- [x] Dry-run produced the expected results
- [ ] The [`to-be-announced`](https://github.com/mozilla/bugbot/labels/to-be-announced) tag added if this is worth announcing

### CODE REVIEW

**Comment by shubham-s-del** [comment]
This is intended to fix this - https://github.com/mozilla/bugbot/issues/2099

**Review by suhaibmujahid — APPROVED** [review]
Thank you, @shubham-s-del!

Could you please link to the issue in the PR description?[^1]

[^1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

**Comment by shubham-s-del** [comment]
> Thank you, @shubham-s-del!
> 
> Could you please link to the issue in the PR description?[1](#user-content-fn-1-03e4ca82a0fef397d784c8c1265a251f)
> 
> Did you do a dry-run test? Once you have done that, please check the "Dry-run produced the expected results" item in the description.
> 
> ## Footnotes
> 1. https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue [↩](#user-content-fnref-1-03e4ca82a0fef397d784c8c1265a251f)

Thanks for the review.
Sorry for the late reply.
added to the description.
Could you help me with steps to do the dry run test?

File: bugbot/rules/security_unhide_dups.py [hunk 1]
```diff
@@ -48,7 +49,7 @@ def bug_handler(bug):
             if (
                 bug["resolution"] != "---"
                 and not bug["groups"]
-                and any(keyword.startswith("sec-") for keyword in bug["keywords"])
+                or any(keyword in SECURITY_KEYWORDS for keyword in bug["keywords"])
```
Comment (suhaibmujahid) [inline bugbot/rules/security_unhide_dups.py:52]: Why did you change `and` to `or` here?

**Review by suhaibmujahid — CHANGES_REQUESTED** [review]
Thank you!

> Could you help me with steps to do the dry run test?

You can do a dry run by running `python -m bugbot.rules.RUE_NAME`, example:
```
python -m bugbot.rules.crash_small_volume
```

Before that you need to create a configuration file. See https://github.com/mozilla/bugbot#running-the-bot-rules

Just make sure to replace `bz_api_key` with your token from Bugzilla.

File: bugbot/rules/security_unhide_dups.py (hunk 1 above)
Comment (shubham-s-del) [inline bugbot/rules/security_unhide_dups.py:52]:
```suggestion
                and any(keyword in SECURITY_KEYWORDS for keyword in bug["keywords"])
```

Comment (shubham-s-del) [reply bugbot/rules/security_unhide_dups.py]: my mistake. corrected.

**Comment by shubham-s-del** [comment]
> Thank you!
> 
> > Could you help me with steps to do the dry run test?
> 
> You can do a dry run by running `python -m bugbot.rules.RUE_NAME`, example:
> 
> ```
> python -m bugbot.rules.crash_small_volume
> ```
> 
> Before that you need to create a configuration file. See https://github.com/mozilla/bugbot#running-the-bot-rules
> 
> Just make sure to replace `bz_api_key` with your token from Bugzilla.


I see a 403 from crash-stats API.
```
requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://crash-stats.mozilla.org/api/SuperSearch/?date=%3E%3D2024-01-04&date=%3C2024-01-11&signature=%5EEMPTY%3A+&signature=%5EOOM+&signature=%3DIPCError-browser+%7C+ShutDownKill&signature=%5Ejava.lang.OutOfMemoryError&_results_number=0&_facets_size=1000
```

Do I need LDAP or phabricator key for this too?

**Comment by suhaibmujahid** [comment]
> I see a 403 from crash-stats API.
> ```
> requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://crash-stats.mozilla.org/api/SuperSearch/?date=%3E%3D2024-01-04&date=%3C2024-01-11&signature=%5EEMPTY%3A+&signature=%5EOOM+&signature=%3DIPCError-browser+%7C+ShutDownKill&signature=%5Ejava.lang.OutOfMemoryError&_results_number=0&_facets_size=1000
> ```

You need a key for crash-stats to run `crash_small_volume`: https://crash-stats.mozilla.org/api/tokens/.

> Do I need LDAP or phabricator key for this too?

No, you don't need them for `security_unhide_dups` or `crash_small_volume`.

**Comment by shubham-s-del** [comment]
> > I see a 403 from crash-stats API.
> > ```
> > requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://crash-stats.mozilla.org/api/SuperSearch/?date=%3E%3D2024-01-04&date=%3C2024-01-11&signature=%5EEMPTY%3A+&signature=%5EOOM+&signature=%3DIPCError-browser+%7C+ShutDownKill&signature=%5Ejava.lang.OutOfMemoryError&_results_number=0&_facets_size=1000
> > ```
> 
> You need a key for crash-stats to run `crash_small_volume`: https://crash-stats.mozilla.org/api/tokens/.
> 
> > Do I need LDAP or phabricator key for this too?
> 
> No, you don't need them for `security_unhide_dups` or `crash_small_volume`.

Thanks

I did a dry-run, gist containing the result = https://gist.github.com/shubham-s-del/1530df5f287c621d863c79ab63c0c317

**Review by suhaibmujahid — APPROVED** [review]
Thank you, @shubham-s-del!

**Pull request merged**